apps/aecore/lib/aecore/naming/naming_state_tree.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Aecore.Naming.NamingStateTree do
+  @moduledoc """
+  Top level naming state tree.
+  """
+  alias Aecore.Naming.Naming
+  alias Aeutil.PatriciaMerkleTree
+
+  @type namings_state() :: Trie.t()
+
+  @spec init_empty() :: Trie.t()
+  def init_empty do
+    PatriciaMerkleTree.new(:naming)
+  end
+
+  @spec put(namings_state(), binary(), Naming.t()) :: namings_state()
+  def put(trie, key, value) do
+    serialized = serialize(value)
+    PatriciaMerkleTree.enter(trie, key, serialized)
+  end
+
+  @spec get(namings_state(), binary()) :: Naming.t()
```

Comment:
if you change to `@spec get(namings_state(), binary()) :: Naming.t() | :none` you should not get some of the suppressed dialyzer warnings.

Reply:
Done